i_statistiken_finalisieren

schmitzdominic opened the pull request:
**DONE:**
- Alle zurück Pfeile auf finish gesetzt (verbessert die Performance)
- Komplette App wieder auf Portrait gesetzt
- Statistik Performance deutlich verbessert
- Alle Charts implementiert
- Time Leiste ist nun Fix und verschwindet nicht mehr beim scrollen
- Logo implementiert (mit weißem Hintergrund)
- Floating Button Text mit Hintergrund versehen
- Diverse Bug fixes

**WICHTIG**
- Das LineChart zeigt nur die letzten 7 Tage an, immer! Das hat momentan den Grund das es sehr aufwändig ist die richtigen Zeiten auszulesen und generisch das Chart anzupassen. Es wurde ein Switch eingefügt (C_Statistik Zeile 222) um es später implementieren zu können.

schmitzdominic committed:
Performance der Back buttons optimiert und angepasst.

schmitzdominic committed:
Komplette App auf Portrait umgebaut

schmitzdominic committed:
Auswahl der Zeit hinzugefügt und die Performance der Statistik verbessert.

schmitzdominic committed:
Bar Chart implementiert.

schmitzdominic committed:
Top 3 Artikel implementiert und einen Bug in der OCR_Manuell beseitigt.

schmitzdominic committed:
PieChart implementiert.

schmitzdominic committed:
PieChart erweitert und sonstige Erweiterungen

- FloatingButton erscheint nun immer beim Tabwechsel
- Time Bar ist nun fixiert.

schmitzdominic committed:
BUGFIX

schmitzdominic committed:
Line Chart implementiert.

schmitzdominic committed:
Layout erweitert

schmitzdominic committed:
Logo geändert.

schmitzdominic committed:
Logo ergänzt

schmitzdominic committed:
Top Facts ergänzt

schmitzdominic committed:
Statistiken ausgelagert und Bugs behoben.

schmitzdominic committed:
Floatings Buttons mit Hintergrund versehen.

schmitzdominic committed:
Artikel Zähler Bug gefixed

schmitzdominic committed:
TimeBar richtig positioniert.

schmitzdominic committed:
Letzte Änderungen

schmitzdominic committed:
Dokumentation ergänzt

schmitzdominic committed:
Dokumentation ergänzt

schmitzdominic committed:
Floating Button hintergrund

Sippl reviewed (changes requested):
Hallo Domi, anbei mein Review (Grundsätzlich funktioniert das meiste, paar Dinge hätte ich aber noch, anbei meine Checkliste):

- Alle zurück pfeile funktionieren Einwandfrei (super gelöst); Jedoch hätte ich bei OCR Manuell hierzu noch einen Verbesserungsvorschlag, wenn du über Bon anzeigen einen Bon bearbeitest und dann auf ausversehen auf zurück klickst, wird alles geänderte verworfen, Das könnte zu Nutzungsfehlern führen, hier wäre eine Meldung sehr sinnvoll
- potrait passt
- Statistik Perfomance ist bei mir echt um Welten besser
- zu den Charts sind mir paar Dinge aufgefallen, die noch nicht zu passen scheinen (z.B. Line Chart schneidet Zahlen ab, Bei Zeitraum Auswahl zeigt er merkwürdiges an etc.) Hier würde es Sinn machen, wenn wir das gemeinsam mal durch gehen
- Time liste von Date Picker => Check
- Logo sieht jetz Mega nice aus, jedoch auf Samsung Geräten ist unser Logo wesentlich kleiner als bei anderen Apps....
- Floating Button, Das mit dem Hintergrund ist gewöhnungsbedürftig, aber passt schon und macht auch Sinn

Vom Code her ist mir nicht wirklich was negatives aufgefallen, da du viel einfach nur an andere Position eingefügt hast (siehe on resumee) oder etwas nur hinzugefügt (this. etc). Bei XML Imageview / Textview fehlt manchmal ne ID (möchte ich aber nicht ankreiden, da es eventuell bewusst weggelassen wurde, außer du hast andere Meinung)
Bzgl. der Statistik, müssen wir das mal durchgehen, da stimmt bei mir etwas nicht.

Die Codestellen, die bzgl. den Kleinigkeiten betroffen wären, würde ich dir dann sagen, wenn wir uns mal absprechen.

schmitzdominic commented:
Hallo Fabi,
Danke für deine Review :)

- Das mit der Message kann ich ändern ;)
- Bitte sag mir endlich welche Auflösung ihr habt? xhdpi? xxhdpi? 
- Das mit dem Abschneiden der Zahlen können wir höchstens ändern wenn wir den Bereich vergrößern, aber selbst dann ist es nicht fehlerfrei.
- Was meinst du mit Zeitraumauswahl zeigt er merkwürdiges an?

schmitzdominic committed:
HOTFIX

- Meldung beim verlassen der Manuell Maske hinzugefügt.
- Zahlenbereich neben dem LineChart vergrößert

schmitzdominic committed:
HOTFIX

- Leiste weiter nach links verschoben
- Zahlen sind nun neben dem LineChart sichtbar

schmitzdominic committed:
ID´s hinzugefügt.

Sippl commented:
Hi Domi,
Über deine Fragen haben wir ja bereits geschrieben. Ich werde alles schnell nochmal checken.
Samsung S5 hat xxhdpi.

Sorry das mit dem Zeitraum war wegen der Anzeige, dass bei zu Hohen Beträgen im BarChart, nur noch die angezeigt werden, die so eine hohe Summe besitzen (haben wir ja beredet nach dem Screenshot)

schmitzdominic committed:
Zurück Bug behoben beim editieren

Sippl reviewed (approved):
Hi Domi,

von meiner Seite gibt es soweit ein Approve:

- Alle Zurück-Pfeile überpüft, funktioneren tadellos
  (bei OCR-manuell über BonAnzeigen => Bei Änderung ohne zu Speichern eine Meldung hinzugefügt)
- Potrait passt
- Statistik Performance klar verbessert
- Charts passen soweit (Die Änderungswünsche (Titel bei Top Produkt sowie Line-Chart Y-Achsenbeschriftung angepasst) hast du ja berücksichtigt, bei dem Bar-Chart das von mir erwähnte Manko ist momentan nicht lösbar)
- Time Leiste in Date Pickern funktioniert super
- Logo sieht sehr gut aus, warum die Größe bei S5 immer noch nicht passt weiß ich nicht
- Floating Button Hintergrundtext ist ne gute Sache
- Fehlende ID's wurden eingefügt

Soweit hatte ich bei von mir angewendeten Anwendung Szenarien keinen Bug bei den Statistiken (außer das man bei zu hohen Beträgen die anderen Beträge nicht mehr sieht :D) Mal schauen vll. findet der Bugmeister Alex noch was 🗡

schmitzdominic committed:
Editier Message hinzugefügt.

schmitzdominic committed:
Dokumentation hinzugefügt.

SemperF1 reviewed (approved):
Wirklich gute Arbeit, gefällt mir gut. Mir ist beim Testen der App nix aufgefallen. Der Code schaut auch gut aus. Ist soweit nix falsch.
